### PR TITLE
v5: [Polls Redesign] Poll Create View

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links

[IOS-1437](https://linear.app/stream/issue/IOS-1437/v5-polls-redesign-poll-create-view)

### 🎯 Goal

Redesign the Create Poll view to match the new v5 design system.

### 📝 Summary

- Redesign `CreatePollView` with new layout, design tokens, and extracted subviews (`CreatePollOptionRow`, `CreatePollSettingCard`, `CreatePollMaxVotesStepper`)
- Refactor `CreatePollViewModel` with `private(set)` access control on `options`, `optionsErrorIndices`, and `maxVotes`, exposing explicit mutation methods
- Replace string-based max votes input with an integer stepper (increment/decrement)
- Add snapshot tests for filled question/options, duplicate option errors, and many options states
- Update `CreatePollViewModel_Tests` to cover all new stepper logic and validation edge cases

### 🛠 Implementation

**Custom Navigation Header**: Replaced `NavigationContainerView` + toolbar with a custom `CreatePollHeader` struct. The confirm button now visually distinguishes enabled (blue `buttonPrimaryBackground`) from disabled (gray `backgroundCoreDisabled`) states, matching the Figma design with 40px visual buttons, 48px hit targets, and 20px icons.

**View Decomposition**: Extracted `CreatePollOptionRow`, `CreatePollSettingCard`, and `CreatePollMaxVotesStepper` as private structs to improve readability and follow SwiftUI composition best practices.

**ViewModel Access Control**: Made `options`, `optionsErrorIndices`, and `maxVotes` `private(set)` so only the ViewModel can mutate them. Added `updateOption(at:value:)`, `removeOption(at:)`, `moveOptions(from:to:)`, and `replaceAllOptions(_:)` for controlled mutations.

### 🧪 Manual Testing Notes

1. Open the Create Poll composer
2. Verify the confirm button appears gray/disabled when no question or options are entered
3. Enter a question and at least one option — confirm button should turn blue
4. Add duplicate options — verify the red "Option already exists" error appears and confirm button becomes disabled
5. Test reordering options via drag handles
6. Test the max votes stepper (increment/decrement) when multiple votes is enabled
7. Verify discard confirmation appears when dismissing with unsaved changes

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [x] New code is covered by unit tests
- [ ] Documentation has been updated in the `docs-content` repo

### CODE REVIEW

File: Sources/StreamChatSwiftUI/ChatComposer/Polls/CreatePollView.swift [hunk 1]
```diff
@@ -54,181 +24,456 @@ public struct CreatePollView: View {
...
+                questionSection
+                optionsSection
+                settingsSpacer
+                settingsSection
                 Spacer()
-                    .modifier(ListRowModifier())
+                    .modifier(CreatePollRowModifier(topSpacing: 0, bottomSpacing: 0))
             }
-            .background(Color(colors.background).ignoresSafeArea())
+            .environment(\.defaultMinListRowHeight, 1)
```
Comment (martinmitrevski) [inline Sources/StreamChatSwiftUI/ChatComposer/Polls/CreatePollView.swift:56]: why do we need this?

**Review by martinmitrevski — COMMENTED** [review]
Looks good! There's one bug: if I have three items, e.g. A1, A2, A3. I move A3 to the first place, the list becomes A3, A1, A3 (A2 goes away), and it shows an error.

File: Sources/StreamChatSwiftUI/ChatComposer/Polls/CreatePollView.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatComposer/Polls/CreatePollView.swift:56]: This is so that the views actually control the height, and not the list items

**Review by martinmitrevski — APPROVED** [review]
